src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt

```diff
@@ -1,39 +1,204 @@
+            e.presentation.isEnabledAndVisible = false
+            return
+        }
+
+        e.presentation.isEnabled = true
+        e.presentation.isVisible = true
+        e.presentation.icon = if (host.isConnectedToUnrealEditor) connectedIcon else disconnectedIcon
+    }
+}
+
+class ProtocolStatus : DumbAwareAction() {
+    override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.BGT
+
+    override fun actionPerformed(e: AnActionEvent) = Unit
+
     override fun update(e: AnActionEvent) {
         super.update(e)
-        e.presentation.isVisible = e.getHost()?.isUnrealEngineSolution ?: false
-        e.presentation.isEnabled = e.getHost()?.isConnectedToUnrealEditor ?: false
+
+        e.presentation.isEnabled = false
+
+        val host = e.getUnrealHost() ?: return
+        e.presentation.isVisible = true
+        e.presentation.text = getActionText(host)
+    }
+
+    @NlsActions.ActionText
+    fun getActionText(host: UnrealHost): String {
+        return if (host.isConnectedToUnrealEditor) {
+            val info = host.connectionInfo
+            if (info != null) {
+                UnrealLinkBundle.message("action.RiderLink.ProtocolStatus.connected.ex.text",
+                        info.projectName, info.executableName, info.processId.toString())
+            } else {
+                UnrealLinkBundle.message("action.RiderLink.ProtocolStatus.connected.text")
+            }
+        } else {
+            UnrealLinkBundle.message("action.RiderLink.ProtocolStatus.disconnected.text")
+        }
+    }
+}
+
+class AttachToConnectedEditor : DumbAwareAction() {
+    override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.EDT
+
+    override fun update(e: AnActionEvent) {
+        super.update(e)
+
+        val host = e.getUnrealHost()
+        if (host == null) {
+            e.presentation.isEnabledAndVisible = false
+            return
+        }
+
+        e.presentation.isVisible = true
+
+        if (!host.isConnectedToUnrealEditor) {
+            e.presentation.isEnabled = false
+            return
+        }
+
+        val connectionInfo = host.connectionInfo
+        if (connectionInfo == null) {
+            e.presentation.isEnabled = false
+            return
+        }
+
+        // try to enumerate current debug sessions and find if we have already a debugger attached
+        /*
+        val debuggerManager = XDebuggerManager.getInstance(host.project)
+        for (session in debuggerManager.debugSessions) {
+            val debugProcess = session.debugProcess
+            if (debugProcess is CidrDebugProcess) {
+                // TODO: need to retrieve target pid from process
+                val pid = 0 // debugProcess.processHandler
+                if (pid == connectionInfo.processId) {
+                    e.presentation.isEnabled = false
+                    return
+                }
+            }
+        }
+        */
```

Comment:
Is it here for the future functionality? I generally don't like dead commented code.

Reply:
And we have the pid of the process now, mb we can uncomment it?

Reply:
Yes, it is the code for the future functionality. We need to enumerate all current debugging sessions and find if we already attached debugger to the same pid as connected by unreal link - we want to disable this action as subsequent attaching will always fail with non-obvious error message (WinAPI things). 
And now there is no way to retrieve target pid from CidrDebugProcess to compare with, and this improvement is not that I want to push to 223 (as ff already happened), so I will implement it in 231. But I don't want to drop code for abovementioned logic, so I commented so someone else can "finish the job" when I'll be not available anymore.